client/reader/search-stream/suggestion-provider.jsx

```diff
@@ -19,36 +20,75 @@ import { getReaderFollowedTags } from 'state/selectors';
  * @return {Array}       An array of suggestions, or null if no tags where provided
  */
 function suggestionsFromTags( count, tags ) {
+	const tagSuggestions = [];
+	let sampleSuggestions = {};
 	if ( tags ) {
 		if ( tags.length <= count ) {
 			return [];
 		}
-		return map( sampleSize( tags, count ), tag => ( tag.displayName || tag.slug ).replace( /-/g, ' ' ) );
+		sampleSuggestions = map( sampleSize( tags, count ), tag => ( tag.displayName || tag.slug ).replace( /-/g, ' ' ) );
+
+		for ( let i = 0; i < sampleSuggestions.length; i++ ) {
```

Comment:
this looks like a `map` to me.  I'd recommend using the original map function like this:
```js
return map(
  sampleSize( tags, count ),
  ( tag, i ) => {
     const text = ( tag.displayName || tag.slug ).replace( /-/g, ' ' );
     return {
       text,
       railcar: {
         railcar: analytics.tracks.createRandomId() + '-' + i,
         ui_algo: 'read:search-suggestions:tags/1',
         ui_position: i,
         rec_result: text,
       }
     }
   }
);
```